flake.nix

```diff
@@ -148,6 +149,17 @@
 
                 wait
               '';
+              nix-cache-proxy = pkgs.writers.writeBashBin "nix-cache-proxy" ''
+
+                nix key generate-secret --key-name foo > skey
```

Comment:
This only should run if the key doesn't exist yet, different secret keys lead to issues.